### PR TITLE
fix: false positives for typescript type + value export

### LINKED ISSUE #1318: import/export: false positives for typescript type + value export
Reporting from https://github.com/typescript-eslint/typescript-eslint/issues/10
The following code is valid typescript (even though it's a bit weird).

```TS
// unexected error: Multiple exports of name 'Foo'. (import/export)
export const Foo = 1;
// unexected error: Multiple exports of name 'Foo'. (import/export)
export type Foo = string;
```

When importing this, it causes the name to work as both a type and a value.
```TS
import { Foo } from './foo';

const x: Foo = 'string val';
const y = Foo; // === 1
```

### PR DESCRIPTION
**Pull request by bradzacher**
Fixes #1318

### CODE REVIEW

**Comment by ljharb** [comment]
This seems like it might overlap with #923?

**Comment by bradzacher** [comment]
@ljharb to prevent double commenting - https://github.com/benmosher/eslint-plugin-import/issues/1318#issuecomment-482433011

File: src/rules/export.js [hunk 1]
```diff
@@ -34,7 +35,14 @@ module.exports = {
         if (node.declaration == null) return
 
         if (node.declaration.id != null) {
-          addNamed(node.declaration.id.name, node.declaration.id)
+          if ([
+            'TSTypeAliasDeclaration',
+            'TSInterfaceDeclaration',
+          ].includes(node.declaration.type)) {
```
Comment (ljharb) [inline src/rules/export.js]: `includes` isn't available on node 4; we'll need to add the `array-includes` package here.

File: tests/src/rules/export.js [hunk 2]
```diff
@@ -106,3 +107,43 @@ ruleTester.run('export', rule, {
...
+    }
+
+    ruleTester.run('export', rule, {
+      valid: [
+        test({
+          code: `
+            export const Foo = 1;
+            export type Foo = number;
+          `,
+          ...parserConfig,
```
Comment (ljharb) [inline tests/src/rules/export.js]: also, spread isn't on node < 8.

**Pull request merged**